domain/application/service/application.go

```diff
@@ -887,12 +893,8 @@ func (s *Service) ChangeApplicationScale(ctx context.Context, appName string, sc
 	}
 	newScale := currentScaleState.Scale + scaleChange
 	s.logger.Tracef(ctx, "ChangeScale DesiredScale %v, scaleChange %v, newScale %v", currentScaleState.Scale, scaleChange, newScale)
-	if newScale < 0 {
-		return -1, internalerrors.Errorf(
-			"%w: cannot remove more units than currently exist", applicationerrors.ScaleChangeInvalid)
-	}
 
-	err = s.st.SetDesiredApplicationScale(ctx, appID, newScale)
+	err = s.st.UpdateApplicationScale(ctx, appID, scaleChange)
```

Comment:
Delete the lines about getting the scale and the adding a newScale. Although, it's kinda good to have a precheck, it might be wrong, so just remove it and do everything in state.

Reply:
Yeah good call, I now return the newScale val from state which was the thing to do in the first place.